src/main/java/net/querz/nbt/ListTag.java

```diff
@@ -31,7 +37,7 @@ public Tag get(int index) {
 	public Tag set(int index, Tag tag) {
 		Tag old = value.get(index);
 		if (!updateType(tag)) {
-			throw new UnsupportedOperationException(String.format("trying to set tag of type %d in ListTag of %d", tag.getID(), type));
+			throw new UnsupportedOperationException(String.format("trying to set tag of type %d in ListTag of %d", tag.getType().id, type.id));
```

Comment:
Those exceptions could show the name of the tag type instead of the ID now, since they have been enumified and we have the tag type:

```java
String.format("trying to set tag of type %s in ListTag of %s", tag.getType(), type)
```